pkg/agent/plugin/svidstore/disk/disk_test.go

```diff
@@ -0,0 +1,362 @@
+MAkGA1UEBhMCVVMxDzANBgNVBAoTBlNQSUZGRTCBmzAQBgcqhkjOPQIBBgUrgQQA
+IwOBhgAEAZ6nXrNctKHNjZT7ZkP7xwfpMfvc/DAHc39GdT3qi8mmowY0/XuFQmlJ
+cXXwv8ZlOSoGvtuLAEx1lvHNZwv4BuuPALILcIW5tyC8pjcbfqs8PMQYwiC+oFKH
+BTxXzolpLeHuFLAD9ccfwWhkT1z/t4pvLkP4FCkkBosG9PVg5JQVJuZJo4GFMIGC
+MA4GA1UdDwEB/wQEAwIBhjAPBgNVHRMBAf8EBTADAQH/MB0GA1UdDgQWBBT4RuNt
+x6E70yjV0wIvUyrGkMKczzAfBgNVHSMEGDAWgBRGyozl9Mjue0Y3w4c2Q+3u+wVk
+CjAfBgNVHREEGDAWhhRzcGlmZmU6Ly9leGFtcGxlLm9yZzAKBggqhkjOPQQDBAOB
+jAAwgYgCQgHOtx4sNCioAQnpEx3J/A9M6Lutth/ND/h8D+7luqEkd4tMrBQgnMj4
+E0xLGUNtoFNRIrEUlgwksWvKZ3BksIIOMwJCAc8VPA/QYrlJDeQ58FKyQyrOIlPk
+Q0qBJEOkL6FrAngY5218TCNUS30YS5HjI2lfyyjB+cSVFXX8Szu019dDBMhV
+-----END CERTIFICATE-----
+`
+)
+
+func TestConfigure(t *testing.T) {
+	for _, tt := range []struct {
+		name            string
+		customConfig    string
+		expectConfig    *configuration
+		expectCode      codes.Code
+		expectMsgPrefix string
+	}{
+		{
+			name:            "no directory provided",
+			expectCode:      codes.InvalidArgument,
+			expectMsgPrefix: "a directory must be configured",
+			expectConfig:    &configuration{},
+		},
+		{
+			name:         "valild config",
+			expectConfig: &configuration{Directory: t.TempDir()},
+			expectCode:   codes.OK,
+		},
+		{
+			name:            "malformed configuration",
+			customConfig:    "{ not a config }",
+			expectCode:      codes.InvalidArgument,
+			expectMsgPrefix: "unable to decode configuration: ",
+			expectConfig:    nil,
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			var err error
+			options := []plugintest.Option{
+				plugintest.CaptureConfigureError(&err),
+			}
+
+			if tt.customConfig != "" {
+				options = append(options, plugintest.Configure(tt.customConfig))
+			} else {
+				options = append(options, plugintest.ConfigureJSON(configuration{
+					Directory: tt.expectConfig.Directory,
+				}))
+			}
+
+			p := New()
+			plugintest.Load(t, builtin(p), nil, options...)
+			spiretest.RequireGRPCStatusHasPrefix(t, err, tt.expectCode, tt.expectMsgPrefix)
+
+			require.Equal(t, tt.expectConfig, p.config)
+		})
+	}
+}
+
+func TestPutX509SVID(t *testing.T) {
+	x509Cert, err := pemutil.ParseCertificate([]byte(x509CertPem))
+	require.NoError(t, err)
+
+	x509Key, err := pemutil.ParseECPrivateKey([]byte(x509KeyPem))
+	require.NoError(t, err)
+
+	x509Bundle, err := pemutil.ParseCertificate([]byte(x509BundlePem))
+	require.NoError(t, err)
+
+	for _, tt := range []struct {
+		name string
+
+		req           *svidstore.X509SVID
+		expectCode    codes.Code
+		expectMsg     string
+		certChainFile string
+		keyFile       string
+		bundleFile    string
+	}{
+		{
+			name:          "Put X509-SVID (successful)",
+			certChainFile: "tls.crt",
+			keyFile:       "tls.key",
+			bundleFile:    "ca.crt",
+		},
+		{
+			name:          "Put X509-SVID (successful)",
+			certChainFile: "tls.crt",
+			keyFile:       "tls.key",
+			bundleFile:    "ca.crt",
+		},
+		{
+			name:          "Put X509-SVID (non existent directory)",
```

Comment:
what do you thing about removing `Put X509-SVID ` part? test will already set TestPutX509SVID as name so, test will be TestPutX509SVID_PutX509-SVID_(non existent_directory), but you can simplify itcalling it `non existent directory`